app/react/portainer/HomeView/EnvironmentList/EnvironmentItem/AgentDetails.tsx

```diff
@@ -1,10 +1,39 @@
+import { AlertTriangle } from 'lucide-react';
+
 import { Environment } from '@/react/portainer/environments/types';
-import { isAgentEnvironment } from '@/react/portainer/environments/utils';
+import {
+  isAgentEnvironment,
+  isEdgeEnvironment,
+} from '@/react/portainer/environments/utils';
+import { isVersionSmaller } from '@/react/common/semver-utils';
+import { useSystemStatus } from '@/react/portainer/system/useSystemStatus';
+
+import { Tooltip } from '@@/Tip/Tooltip';
+import { Icon } from '@@/Icon';
 
 export function AgentDetails({ environment }: { environment: Environment }) {
-  if (!isAgentEnvironment(environment.Type)) {
+  const { data: systemStatus } = useSystemStatus();
+
+  if (!systemStatus || !isAgentEnvironment(environment.Type)) {
     return null;
   }
 
-  return <span>{environment.Agent.Version}</span>;
+  const { Version } = systemStatus;
+  const isSmallerEdge =
+    environment.Agent.Version &&
+    isVersionSmaller(environment.Agent.Version, Version) &&
+    isEdgeEnvironment(environment.Type);
+
+  return (
```

Comment:
what about showing the version for non-edge, and if it's edge, have another component which reaches the server for the server version?

Reply:
That's what it is doing, will spit in 2 components

Reply:
yeah, I meant just for edge. it's good this way too